agenta-backend/agenta_backend/routers/observability_router.py

```diff
@@ -1,99 +1,163 @@
-import os
-from typing import List
+from typing import List, Any
 
-from fastapi import Request
-from agenta_backend.utils.common import APIRouter
+from fastapi import Request, Query, Depends
 
-from agenta_backend.services.event_db_manager import (
-    get_variant_traces,
-    create_app_trace,
-    create_trace_span,
-    get_trace_single,
-    trace_status_update,
-    get_trace_spans,
-    add_feedback_to_trace,
-    get_trace_feedbacks,
-    get_feedback_detail,
-    update_trace_feedback,
+from agenta_backend.utils.common import APIRouter
+from agenta_backend.services import event_db_manager, redis_cache_service
+from agenta_backend.models.api.api_models import (
+    WithPagination,
+    SorterParams,
+    PaginationParam,
 )
 from agenta_backend.models.api.observability_models import (
-    Span,
+    SpanDetail,
     CreateSpan,
     CreateFeedback,
     Feedback,
     UpdateFeedback,
-    Trace,
+    TraceDetail,
     CreateTrace,
     UpdateTrace,
+    GenerationFilterParams,
+    ObservabilityDashboardDataRequestParams,
 )
 
 
 router = APIRouter()
 
 
+@router.get(
+    "/dashboard/",
+    operation_id="observability_dashboard",
+)
+async def get_dashboard_data(
+    request: Request,
+    app_id: str,
+    parameters: ObservabilityDashboardDataRequestParams = Depends(),
+):
+    try:
+        dashboard_data = await redis_cache_service.cache_observability_data(
```

Comment:
I think we should not cache requests about the last 24 hours. Otherwise the dashboard will always look empty if we open it at the beginning of the day, then start working. No?

Reply:
We aren't caching in the last 24 hours. We are caching every 15 minutes.